src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -302,6 +304,17 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
         error.addToHolder(holder)
     }
 
+    private fun checkUnstableAttribute(ref: RsReferenceElement, o: RsElement, holder: RsAnnotationHolder) {
+        val element = ref.reference.resolve() as? RsOuterAttributeOwner ?: return
+        for (attr in element.queryAttributes.unstableAttributes) {
+            val metaItems = attr.metaItemArgs?.metaItemList ?: continue
+            val featureName = metaItems.singleOrNull { it.name == "feature" }?.value ?: continue
+            val reason = metaItems.singleOrNull { it.name == "reason" }?.value?.let(::unescapeStringCharacters)
+            val feature = CompilerFeature.find(featureName) ?: CompilerFeature(featureName, FeatureState.ACTIVE)
+            feature.check(holder, o, null, "`$featureName` is unstable${if (reason != null) ": $reason" else ""}")
```

Comment:
nit: I'd would extract `${if (reason != null) ": $reason" else ""}` to `reasonSuffix` variable. Looks like it's a bit more readable